rskj-core/src/test/java/co/rsk/peg/union/UnionBridgeIT.java

```diff
@@ -0,0 +1,153 @@
+import static co.rsk.peg.BridgeMethods.REQUEST_UNION_BRIDGE_RBTC;
+import static co.rsk.peg.BridgeMethods.SET_UNION_BRIDGE_CONTRACT_ADDRESS_FOR_TESTNET;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import co.rsk.core.Coin;
+import co.rsk.core.RskAddress;
+import co.rsk.peg.Bridge;
+import co.rsk.peg.BridgeIllegalArgumentException;
+import co.rsk.peg.BridgeSupport;
+import co.rsk.peg.constants.BridgeMainNetConstants;
+import co.rsk.peg.union.constants.UnionBridgeConstants;
+import co.rsk.test.builders.BridgeBuilder;
+import co.rsk.test.builders.BridgeSupportBuilder;
+import co.rsk.test.builders.UnionBridgeSupportBuilder;
+import java.math.BigInteger;
+import org.ethereum.TestUtils;
+import org.ethereum.config.blockchain.upgrades.ActivationConfig;
+import org.ethereum.config.blockchain.upgrades.ActivationConfigsForTest;
+import org.ethereum.core.CallTransaction;
+import org.ethereum.core.SignatureCache;
+import org.ethereum.core.Transaction;
+import org.ethereum.vm.exception.VMException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class UnionBridgeIT {
+
+    private static final ActivationConfig allActivations = ActivationConfigsForTest.all();
+    private static final ActivationConfig lovellActivations = ActivationConfigsForTest.lovell700();
+
+    private static final BridgeMainNetConstants bridgeMainNetConstants = BridgeMainNetConstants.getInstance();
+    private static final UnionBridgeConstants unionBridgeMainNetConstants = bridgeMainNetConstants.getUnionBridgeConstants();
+
+    private static final UnionBridgeSupportBuilder unionBridgeSupportBuilder = UnionBridgeSupportBuilder.builder();
+
+    private static final BridgeSupportBuilder bridgeSupportBuilder = BridgeSupportBuilder.builder();
+    private BridgeSupport bridgeSupport;
+
+    private static final BridgeBuilder bridgeBuilder = new BridgeBuilder();
+    private Bridge bridge;
+
+    private static final RskAddress CURRENT_UNION_BRIDGE_ADDRESS = new RskAddress(
+        "5988645d30cd01e4b3bc2c02cb3909dec991ae31");
+    private static final RskAddress NEW_UNION_BRIDGE_CONTRACT_ADDRESS = TestUtils.generateAddress(
+        "newUnionBridgeContractAddress");
+
+    private static final int LOCKING_CAP_INCREMENTS_MULTIPLIER = unionBridgeMainNetConstants.getLockingCapIncrementsMultiplier();
+    private static final Coin NEW_LOCKING_CAP = unionBridgeMainNetConstants.getInitialLockingCap()
+        .multiply(BigInteger.valueOf(
+            LOCKING_CAP_INCREMENTS_MULTIPLIER));
+
+    private static final BigInteger ONE_ETH = BigInteger.TEN.pow(
+        18); // 1 ETH = 1000000000000000000 wei
+    private static final Coin AMOUNT_TO_REQUEST = new co.rsk.core.Coin(ONE_ETH);
+    private static final Coin AMOUNT_TO_RELEASE = new co.rsk.core.Coin(
+        ONE_ETH.multiply(BigInteger.TEN));
```

Comment:
Maybe not very significant at this stage, but why is the release value higher than the request?

Reply:
Updated